interview-questions.md

```diff
@@ -4,49 +4,64 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. Rails follows an MVC pattern. What does that mean?
 
-  Your answer:
+  Your answer: I forget to be honest!
 
-  Researched answer:
+  Researched answer: MVC is a pattern for the architecture of a software application. It separates an application into the following components:
+
+  Models for handling data and business logic
+  Controllers for handling the user interface and application
+  Views for handling graphical user interface objects and presentation
 
 
 
 2. What does it mean if an application has CRUD? What does CRUD stand for?
 
-  Your answer:
+  Your answer: The four basic principles for a good application are CRUD - Create, Read, Update, Delete - meaning a user should be able to create data, read what they've created, update it if needed, then delete it.
 
-  Researched answer:
+  Researched answer: Create, read, update, and delete[1] (CRUD) are the four basic functions of persistent storage. Without at least these four operations, the software cannot be considered complete.
+
+  - create or add new entries;
+  - read, retrieve, search, or view existing entries;
+  - update or edit existing entries;
+  - delete, deactivate, or remove existing entries.
 
 
 
 3. What is a migration? Why would you use one?
 
-  Your answer:
+  Your answer: A migration updates the schema of a database, so you don't have to interact with it directly. You'll have record of each migration
```

Comment:
yes! you want a record of all the db changes rather that altering the schema directly